dlpy/layers.py

```diff
@@ -114,17 +115,46 @@ class Layer(object):
     def __init__(self, name=None, config=None, src_layers=None):
         self.name = name
         self.config = config
+        self.depth = None
 
         if src_layers is None:
             self.src_layers = None
         else:
-            self.src_layers = list(src_layers)
+            # to be compatible with network
+            if isinstance(src_layers, collections.Iterable):
+                self.src_layers = list(src_layers)
+            else:
+                self.src_layers = [src_layers]
 
         if 'act' in self.config.keys() and self.config['act'] is not None:
             self.activation = self.config['act'].title()
         else:
             self.activation = None
 
+    def __call__(self, inputs, **kwargs):
+        layer_type = self.__class__.__name__
+        if isinstance(inputs, list) and len(inputs) > 1:
+            if layer_type not in ['Concat', 'Res', 'Scale', 'Dense']:
+                raise DLPyError('The input of {layer_type} should have only one layer.')
```

Comment:
{layer_type} needs to be interpolated here.  Right now, it will just show up literally in the error message.